packages/contracts-bedrock/src/L1/DataAvailabilityChallenge.sol

```diff
@@ -0,0 +1,343 @@
+///         commitment was made.
+///         If the challenge is not resolved within a certain block interval (resolveWindow), the challenge can be
+///         expired.
+///         If a challenge is expired, the challenger's bond is unlocked and the challenged commitment is added to the
+///         chain of expired challenges.
+contract DataAvailabilityChallenge is OwnableUpgradeable, ISemver {
+    /// @notice Error for when the provided resolver refund percentage exceeds 100%.
+    error InvalidResolverRefundPercentage(uint256 invalidResolverRefundPercentage);
+
+    /// @notice Error for when the challenger's bond is too low.
+    error BondTooLow(uint256 balance, uint256 required);
+
+    /// @notice Error for when attempting to challenge a commitment that already has a challenge.
+    error ChallengeExists();
+
+    /// @notice Error for when attempting to resolve a challenge that is not active.
+    error ChallengeNotActive();
+
+    /// @notice Error for when attempting to unlock a bond from a challenge that is not expired.
+    error ChallengeNotExpired();
+
+    /// @notice Error for when attempting to challenge a commitment that is not in the challenge window.
+    error ChallengeWindowNotOpen();
+
+    /// @notice Error for when the provided input data doesn't match the commitment.
+    error InvalidInputData(bytes32 providedDataHash, bytes32 expectedHash);
+
+    /// @notice Error for when the call to withdraw a bond failed.
+    error WithdrawalFailed();
+
+    /// @notice An event that is emitted when the status of a challenge changes.
+    /// @param challengedHash The hash of the commitment that is being challenged.
+    /// @param challengedBlockNumber The block number at which the commitment was made.
+    /// @param status The new status of the challenge.
+    event ChallengeStatusChanged(
+        bytes32 indexed challengedHash, uint256 indexed challengedBlockNumber, ChallengeStatus status
+    );
+
+    /// @notice An event that is emitted when the bond size required to initiate a challenge changes.
+    event RequiredBondSizeChanged(uint256 challengeWindow);
+
+    /// @notice An event that is emitted when the percentage of the resolving cost to be refunded to the resolver
+    /// changes.
+    event ResolverRefundPercentageChanged(uint256 resolverRefundPercentage);
+
+    /// @notice An event that is emitted when a user's bond balance changes.
+    event BalanceChanged(address account, uint256 balance);
+
+    /// @notice Semantic version.
+    /// @custom:semver 0.0.0
+    string public constant version = "0.0.0";
+
+    /// @notice The fixed cost of resolving a challenge.
+    uint256 public constant fixedResolutionCost = 44200;
+
+    /// @notice The variable cost of resolving a callenge per byte of calldata.
+    /// @dev upper limit; 16 gas per non-zero calldata byte, 4 gas variable execution cost per byte.
+    uint256 public constant variableResolutionCost = 16 + 4;
```

Comment:
These parameters need to be tuned. They're currently based on measurements with a forge test, but it's hard to get realistic measurements for calldata cost in a forge test, so we're working on a ts/viem script that deploys the contract on a local anvil and then challenges/resolves with a bunch of random data to verify these numbers.

Reply:
updated after running an EOA script to estimate the costs